WMF Framework/ExploreCardCollectionViewCell.swift

```diff
@@ -226,24 +225,30 @@ public class ExploreCardCollectionViewCell: CollectionViewCell, Themeable {
                 cardBackgroundView.frame = cardContentViewFrame.insetBy(dx: -singlePixelDimension, dy: -singlePixelDimension)
             }
             origin.y += cardContentViewFrame.layoutHeight(with: 20)
-        } else if isCollapsed {
+        }
+
+        if isCollapsed, !undoLabel.isHidden, !undoButton.isHidden {
+            let undoInsets: (x: CGFloat, y: CGFloat) = (x: 15, y: 16)
+            labelOrigin.x += undoInsets.x
+            labelOrigin.y += undoInsets.y
+
+            let undoLabelMaxWidth = widthMinusMargins - (widthMinusMargins * 0.25)
+            let undoLabelMinWidth = widthMinusMargins * 0.5
+            let undoLabelX = isRTL ? widthMinusMargins - undoLabelMaxWidth : labelOrigin.x
+            let undoLabelFrameHeight = undoLabel.wmf_preferredHeight(at: CGPoint(x: undoLabelX, y: labelOrigin.y), maximumWidth: undoLabelMaxWidth, minimumWidth: undoLabelMinWidth, horizontalAlignment: labelHorizontalAlignment, spacing: 0, apply: apply)
+
+            let undoButtonMaxWidth = widthMinusMargins * 0.25
+            let undoButtonX = isRTL ? labelOrigin.x : widthMinusMargins - undoButtonMaxWidth
+            let undoButtonFrameHeight = undoButton.wmf_preferredHeight(at: CGPoint(x: undoButtonX, y: labelOrigin.y), maximumWidth: undoButtonMaxWidth, horizontalAlignment: buttonHorizontalAlignment, spacing: 0, apply: apply)
+
+            let undoHeight = max(undoLabelFrameHeight, undoButtonFrameHeight)
+            let cardBackgroundViewHeight = ceil(origin.y) + undoHeight + undoInsets.y
+            let cardBackgroundViewFrame = CGRect(x: layoutMargins.left, y: layoutMargins.top, width: widthMinusMargins, height: cardBackgroundViewHeight)
             if apply {
-                let cardBackgroundViewFrame = CGRect(x: layoutMargins.left, y: contentView.frame.origin.y, width: widthMinusMargins, height: contentView.frame.height)
                 cardBackgroundView.frame = cardBackgroundViewFrame
             }
-        }
-
-        if !undoLabel.isHidden {
-            let undoLabelOrigin = CGPoint(x: labelOrigin.x + 14, y: cardBackgroundView.frame.midY)
-            var undoLabelFrame = undoLabel.wmf_preferredFrame(at: undoLabelOrigin, maximumWidth: widthMinusMargins, minimumWidth: 84, horizontalAlignment: labelHorizontalAlignment, apply: false)
-            let halfHeight = round(0.5 * undoLabelFrame.height)
-            undoLabelFrame.origin.y -= halfHeight
-            undoLabel.frame = undoLabelFrame
-        }
 
-        if !undoButton.isHidden {
-            let undoButtonOrigin = CGPoint(x: origin.x - 8, y: undoLabel.frame.origin.y)
-            origin.y += undoButton.wmf_preferredHeight(at: undoButtonOrigin, maximumWidth: widthMinusMargins, horizontalAlignment: buttonHorizontalAlignment, spacing: 20, apply: apply)
+            origin.y += cardBackgroundViewFrame.layoutHeight(with: layoutMargins.bottom)
```

Comment:
I think we want to add `layoutMargins.bottom` if it isn't collapsed as well. Right now the space is made up for with `let interItemSpacing: CGFloat = 32` in layout metrics, which should really be `17` and the `layoutMargins.bottom` is the remaining`15`

Reply:
what about 'layoutMargins.top'? If `interItemSpacing` becomes `17`, wouldn't we need to add `layoutMargins.top` to have `32` in spacing on both ends?

Reply:
`interItemSpacing` is just between the items. If we want additional spacing above the first item or below the last item, we would need to add `sectionInset`. This naming convention matches `UICollectionViewFlowLayout`. My implementation in `ColumnarCollectionViewLayoutInfo` is a bit off at the moment but should probably be updated. For example, it's currently erroneously adding `interItemSpacing` below the last item